src/com/palmergames/bukkit/util/Version.java

```diff
@@ -35,14 +37,23 @@ public Version(String version) {
 		return 0;
 	}
 
-	@Override public boolean equals(Object that) {
-		if(this == that)
-			return true;
-		if(that == null)
-			return false;
-		if(this.getClass() != that.getClass())
-			return false;
-		return this.compareTo((Version) that) == 0;
+	@Override
+	public boolean equals(Object o) {
+		if (this == o) return true;
+		if (!(o instanceof Version)) return false;
+		Version version1 = (Version) o;
+		return Objects.equals(version, version1.version);
```

Comment:
I think this has the possibility of being recursive since `Object.equals()` internally calls `.equals()`. At the least it's a useless call.

Reply:
It's calling it on the field version not the object itself.